### PR TITLE
Improve the alignment of different buttons(list, calendar, starred,

### PR DESCRIPTION
**Pull request by Princu7**
search) along with full utilization of screen space.

**Fixes** #1265 

**Changes**:
* All the different buttons(list, view, calendar and search bar) are aligned in a single row and the full width of the screen is utilized for showing the date navbar and the buttons.
* Change "Show Starred" to "Starred" and "Show All" to "All".
* The buttons are responsive and behave nicely on different screen sizes.
* Removes the redundant and extra classes written for the search and the filter bar.

**Screenshots for the change**: 

* Schedule Page
screenshot from 2017-05-13 07-54-55

* Rooms Page
screenshot from 2017-05-13 07-55-10

**Demo Link**

* Event Site  : http://princu7.github.io/eventSiteWebApp
* Server : http://secure-meadow-20680.herokuapp.com

@mariobehling @aayusharora @geekyd Please Review. Thanks!!

### CODE REVIEW

**Comment by mariobehling** [comment]
Thanks, but please adhere to former design decisions. I see this happening many times recently, when there is a UI enhancement - at the same time border distances, spaces alignments or something small gets changed resulting in another PR, where it is fixed and then something else is changed. How can we improve the quality of our process?

If I ask, someone will say point a) is true, but point b) was changed elsewhere. It is not my fault. Someone else did it. If you work in an area, please ensure common design practices are followed, e.g. distance to borders, alignments, common heights and design of buttons etc. Please see the screenshot below. What do you find you can improve in this PR?
9f85979e-37b1-11e7-82b7-071696137f97

**Comment by mariobehling** [comment]
@Princu7 It would be great if you could start to see these things yourself. If I would open issues now, it would be too much micro-management. The time to open an issue is already much longer than to fix each small item. The main goal is: Can you please see these kind of things and follow common design practices, please?

File: src/backend/_scss/application.scss [hunk 1]
```diff
@@ -1680,7 +1695,9 @@ a.skip:hover {
   background-color:$white-gray;
   color:black;
   border-color:$light-black;
-  margin : 5px 5px 10px 5px;
+  margin : 0px;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: px in 0 Margin, Remove please.......

File: src/backend/_scss/application.scss [hunk 2]
```diff
@@ -1386,36 +1389,48 @@ a {
...
-  float: right;
+
+.search-filter-schedule {
+  flex-grow: 1;
   padding-top: 12px;
 
   .fossasia-filter {
-    float: left;
-    font-family: sans-serif;
+    height: 70%;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Why it is in percentages. Do you want to keep it with respect to parent?

File: src/backend/_scss/application.scss [hunk 3]
```diff
@@ -1337,17 +1345,14 @@ a {
...
-.search-filter,
-.search-filter-speaker {
+.search-filter {
+
   .fossasia-filter {
     background-color: $white;
     border: 1px solid $gray-dim;
     color: $black;
-    font-family: verdana sans-serif;
+    font-family: sans-serif;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Any reason for changing font ?

File: src/backend/_scss/application.scss [hunk 4]
```diff
@@ -411,8 +411,18 @@ a {
 /* Track SCSS */
 
 .show-button-filter {
-  display: inline-block;
-  padding-top: 6px;
+  padding-top: 10px;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Are you sure going to Flex-box approach. I doubt what will happen when Date-filters are increased.
Flex will try to make it in single line which makes the font on mobile invisible.

File: src/backend/_scss/application.scss [hunk 5]
```diff
@@ -1386,36 +1389,48 @@ a {
...
   }
 
 }
 
+.filter-margin {
+  margin-right: 10px;
+}
+
+.navbar-flex {
+  display: flex;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Choose flexbox carefully with a reason.

**Comment by aayusharora** [comment]
@Princu7  
The points by @mariobehling are very valid. Can we start with selenium testing asap?

**Comment by Princu7** [comment]
@mariobehling @aayusharora I am sorry. There were a lot of small things to improve and I failed to see the border issue already existing in the app. I will try to be more careful from next time.
Here are the updated screenshots with comparison

**Schedule page**

* Before
screenshot from 2017-05-13 17-27-47
screenshot from 2017-05-13 17-28-20
* After
screenshot from 2017-05-13 17-34-05
screenshot from 2017-05-13 17-34-10

**Comment by Princu7** [comment]
**Tracks page**
* Before
screenshot from 2017-05-13 17-29-18
screenshot from 2017-05-13 17-29-41
* After
screenshot from 2017-05-13 17-34-24
screenshot from 2017-05-13 17-34-28

**Comment by Princu7** [comment]
@aayusharora I have removed the font-size property of the search bar to set the font to the default font used in the web app. We can start with testing soon. As soon as this issue is fixed, we can go ahead with it. Thanks!!

File: src/backend/_scss/application.scss (hunk 5 above)
Comment (Princu7) [inline src/backend/_scss/application.scss]: I tried other approaches but flexbox seemed to work the best for all screen sizes!! The properties like flex-grow and flex-wrap are very useful. That's why I went ahead with it. Thanks!!

File: src/backend/_scss/application.scss (hunk 4 above)
Comment (Princu7) [inline src/backend/_scss/application.scss]: When date filters are increased, it seems to be adjusting well and is in accordance with the current implementation. Please guide. Thanks!!

**Comment by geekyd** [comment]
@Princu7 Please check the Dates for calendar view and list view it seems to have changed, please make its position same for both views.

**Comment by Princu7** [comment]
@geekyd Thanks. I will fix it :)

**Comment by Princu7** [comment]
@mariobehling @aayusharora Sorry. There were many small changes to be done on this issue and I failed to see the existing border problem in the app. I will try to be more careful from now. I have made some progress. I am attaching many screenshots below for easier reviewing process.

screenshot from 2017-05-13 18-29-56

screenshot from 2017-05-13 18-30-08

screenshot from 2017-05-13 18-30-28

screenshot from 2017-05-13 18-30-38

screenshot from 2017-05-13 18-30-43

**Deployment Link**: https://princu7.github.io/eventSiteWebApp.
Please review. Thanks!!

**Comment by aayusharora** [comment]
@Princu7 Ok, Let's go with Flexbox, for now, I have seen you have used `flex-wrap` which is good, just need to check how it wraps in case the number of dates will increase to 6 or more than 6.
Can you please check and provide the result?

**Comment by Princu7** [comment]
@aayusharora Thanks for pointing out the issue. In both the PR implementation and the existing implementation, on adding more date-filters, they overflow on the mobile and the whole site becomes scrollable. To avoid that, I added the `overflow: auto` property and now only the date filter component is scrollable. Here are some screenshots to demonstrate this:

* PR Implementation
screenshot from 2017-05-13 19-01-39

* Existing Implementation(same)
screenshot from 2017-05-13 19-03-06

* After adding `overflow: auto`
screenshot from 2017-05-13 19-02-23

**Comment by aayusharora** [comment]
Thank you @Princu7. Perfect. 
But, the same case should be handled on Desktop too. In case we have more than 6 dates which ( if we accommodate in a single row in flexbox) would produce small text which would be difficult to read.
We can make a separate issue for that.
Thanks.

**Comment by aayusharora** [comment]
@mariobehling Approve and Merge, please
LGTM.

**Comment by geekyd** [comment]
@Princu7 Great job! There is one thing that needs to be fixed please check the screenshot. for tablet view (1024x768). The size of the stared button is too large.

screenshot from 2017-05-13 22-43-39

**Comment by Princu7** [comment]
@geekyd Thanks!! I have updated the PR. It should be fine now.

![screenshot from 2017-05-13 23-23-17]
(https://cloud.githubusercontent.com/assets/8847265/26027860/3c328594-3833-11e7-83a8-b8d28dd493c6.png)

**Comment by mariobehling** [comment]
Ok, let's iterate from here.

**Pull request merged**